bin/reth/src/commands/stage/run.rs

```diff
@@ -99,7 +99,7 @@ impl Command {
                 provider_factory.db_ref().clone(),
                 provider_factory.static_file_provider(),
                 metrics_process::Collector::default(),
-                ctx.task_executor,
+                TaskExecutor::current(),
```

Comment:
Is this something we should do inside the functions to simplify how they get called? I always found it confusing you had to pass a task executor down, and given the tight coupling with Tokio I feel it's fine?